content/article/AMDGPU_20170624.md

```diff
@@ -0,0 +1,110 @@
+
+# AMDGPU-PRO
+
+There have been a few questions I've noticed about having issues with HDMI
+audio over AMDGPU and/or people asking when AMDGPU-PRO will be be supported.
+I'm here to say AMDGPU-PRO isn't coming to Gentoo/Sabayon, but for good reason.
+
+Differences in AMDGPU and AMDGPU-PRO, once complete, will be minimal. 
+AMDGPU-PRO is a "hybrid" driver meaning paritially open and closed source.
+Nearly everything in AMDGPU-PRO will be in AMDGPU open source. AMDGPU-PRO
+will be geared for supporting FirePRO cards in workstations and tested
+to support things such as solidworks and other business software. 
+
+Currently, there are some gaps in support. Where PRO supports Freesync, HDMI 2.0,
+HDMI Audio, OpenCL, etc., the AMDGPU driver will eventually attain all that
+through what is known as Display Code (DC) or DAL/HAL (Display/Hardware abstraction
+layer).  The old DAL was just as it sounds, an abastraction layer. This 
+could basically be summed up as a middle man layer that needed translation
+to be native or direct communication with the hardware. This is not allowed in
+the Linux Kernel and for good reason. We would want native and direct communication
+with hardware which is better for performance and being maintained and clean.
+
+So why did AMD use a HAL? Well, the simplest reason was for sharing code.
+AMD's software driver team is small. To save time, money, and effort, using
+a HAL was a simple method. They could write code that (mostly) worked for
+both Windows and Linux at the same time. So primary focus on Windows translated
+to some focus on Linux. Unfortunately, this doesn't mean it worked well.
+We all remember how 'lovely' fglrx was a times with broken crossfire support,
+broken eyefinity setup, poor kernel support, etc.
+
+OK so small software dev team, HAL isn't fully working in this instance,
+what is the solution? They decided to go open source. "Great, shouldn't this
+be simple?" Short answer, No. They have to go over code with a fine tooth comb
+to make sure they aren't releasing code they "don't own" persay. Can't exactly
+opensource DirectX related etc. After that, they also have to remove the HAL and recreate
+code from scratch to make a native/direct communication line for the linux kernel.
+You could say this is essentially rewriting the driver from scratch. Its also
+more than just a display driver. You have Vulkan, OpenCL, HSA, VCE, UVD and the many 
+other interfaces in which one may use the graphics card. 
+
+So when can we expect to see support for all items the display code brings us?
+Thats an excellent, but daunting question. 
+AMD is about to meet the crossroads in which they either: 
+* Do not support their latest graphics cards at all ,VEGA and beyond, upon release
+* Push their team to get the DC conversion completed and submitted to the linux kernel
```

Comment:
"Push their team to get the display code converted and submitted to the Linux kernel"

Better but now it reads like a generic display code (so why would it need converting?), not *that* DC stat stands next to HAL and DAL.